docs/tools/lsp-factoryjs/classes/02-lsp4-digital-asset-metadata.md

```diff
@@ -0,0 +1,153 @@
+---
+sidebar_position: 2
+title: LSP4DigitalAssetMetadata
+---
+
+# LSP4DigitalAssetMetadata
+
+## uploadMetadata
+
+```js
+LSP4DigitalAssetMetadata.uploadMetadata(lsp4Metadata, uploadOptions?);
+```
+
+The function uploads and processes passed assets and images, and uploads the [LSP4 Digital Asset Metadata](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-4-DigitalAsset-Metadata.md) to IPFS.
+
+The `uploadMetadata` function is available as a static or non-static function callable on the `LSPFactory` library instance. <br/> If `uploadOptions` are not specified in the function call, and the function is used on an `LSPFactory` instance, <br/> the specified options in `uploadOptions` that were passed to the LSPFactory will be used.
+
+#### Parameters
+
+| Name             | Type   | Description                                            |
+| :--------------- | :----- | :----------------------------------------------------- |
+| `metaData`       | Object | The metadata to be uploaded.                           |
+| `uploadOptions?` | Object | The specification how the metadata should be uploaded. |
+
+#### Parameters of `metaData`
+
+| Name          | Type                                             | Description                         |
+| :------------ | :----------------------------------------------- | :---------------------------------- |
+| `description` | string                                           | A description of the digital asset. |
+| `links?`      | {&nbsp;title: string, url: string&nbsp;}[&nbsp;] | The links of the digital asset.     |
+| `icon?`       | File, AssetBuffer, or AssetMetadata[&nbsp;]      | The icon of the digital asset.      |
+| `images?`     | File, AssetBuffer, or AssetMetadata[&nbsp;]      | The images of the digital asset.    |
+| `assets?`     | File, AssetBuffer, or AssetMetadata[&nbsp;]      | The assets of the digital asset.    |
+
+#### Parameters of `uploadOptions?`
+
+| Name                 | Type   | Description                                                                       |
+| :------------------- | :----- | :-------------------------------------------------------------------------------- |
+| `ipfsClientOptions?` | Object | IPFS Client Options as defined by the [ipfs-http-client library] used internally. |
+
+#### Returns
+
+| Name      | Type                          | Description                               |
+| :-------- | :---------------------------- | :---------------------------------------- |
+| `Promise` | &ltLSP4MetadataForEncoding&gt | The processed [LSP4] data and upload URL. |
```

Comment:
Id recommend just writing `<LSP4MetadataForEncoding>` as `&ltLSP4MetadataForEncoding&gt` is confusing to read. Best to avoid mixing HTML and markdown if we can

Reply:
cannot do it like that because it will count as an opening tag

Reply:
You can use \ to ignore the html, e.g. <LSP4MetadataForEncoding\>. This is already done elsewhere in the docs e.g. https://github.com/lukso-network/docs/blob/main/docs/tools/lsp-factoryjs/classes/lsp3-universal-profile.md?plain=1#L46